packages/block-library/src/image/image.js

```diff
@@ -87,28 +87,25 @@ export default function Image( {
 	onSelectURL,
 	onUploadError,
 	containerRef,
+	clientId,
 } ) {
 	const captionRef = useRef();
 	const prevUrl = usePrevious( url );
-	const { block, currentId, image, multiImageSelection } = useSelect(
+	const { getBlock } = useSelect( blockEditorStore );
+	const { image, multiImageSelection } = useSelect(
 		( select ) => {
 			const { getMedia } = select( coreStore );
-			const {
-				getMultiSelectedBlockClientIds,
-				getBlockName,
-				getSelectedBlock,
```

Comment:
Almost feel like we should deprecate this selector or have some kind of warning in all selectors that returns block objects.

Reply:
The two other usage we have of that selectors seem safe.